ql/src/test/queries/clientpositive/sketches4.q

```diff
@@ -0,0 +1,50 @@
+-- add jar /home/dev/hive/packaging/target/apache-hive-4.0.0-SNAPSHOT-bin/apache-hive-4.0.0-SNAPSHOT-bin/lib/datasketches-hive-1.1.0-incubating-SNAPSHOT.jar;
+-- add jar /home/dev/hive/packaging/target/apache-hive-4.0.0-SNAPSHOT-bin/apache-hive-4.0.0-SNAPSHOT-bin/lib/sketches-core-0.9.0.jar;
+-- add jar /home/dev/hive/packaging/target/apache-hive-4.0.0-SNAPSHOT-bin/apache-hive-4.0.0-SNAPSHOT-bin/lib/datasketches-java-1.2.0-incubating.jar;
+-- add jar /home/dev/hive/packaging/target/apache-hive-4.0.0-SNAPSHOT-bin/apache-hive-4.0.0-SNAPSHOT-bin/lib/datasketches-memory-1.2.0-incubating.jar
+
+
+set hive.support.concurrency=true;
+set hive.txn.manager=org.apache.hadoop.hive.ql.lockmgr.DbTxnManager;
+set hive.strict.checks.cartesian.product=false;
+set hive.stats.fetch.column.stats=true;
+set hive.materializedview.rewriting=true;
+set hive.fetch.task.conversion=none;
+
+create table sketch_input (id int, category char(1))
+STORED AS ORC
+TBLPROPERTIES ('transactional'='true');
+
+insert into table sketch_input values
+  (1,'a'),(1, 'a'), (2, 'a'), (3, 'a'), (4, 'a'), (5, 'a'), (6, 'a'), (7, 'a'), (8, 'a'), (9, 'a'), (10, 'a'),
+  (6,'b'),(6, 'b'), (7, 'b'), (8, 'b'), (9, 'b'), (10, 'b'), (11, 'b'), (12, 'b'), (13, 'b'), (14, 'b'), (15, 'b')
+; 
+
+-- build sketches per category
+create table sketch_intermediate (category char(1), sketch binary)
+STORED AS ORC
+TBLPROPERTIES ('transactional'='true');
+ 
+-- create an mv for the intermediate results
+create  materialized view mv_1 as
+  select category, datatosketch2(id),count(id) from sketch_input group by category;
+
+-- see if we use the mv
+explain
+select category, SketchToEstimate2(datatosketch2(id)) from sketch_input group by category;
+
+select category, SketchToEstimate2(datatosketch2(id)) from sketch_input group by category;
+
+-- union sketches across categories and get overall unique count estimate
+explain
+select SketchToEstimate2(datatosketch2(id)) from sketch_input group by category;
+select SketchToEstimate2(datatosketch2(id)) from sketch_input group by category;
+
+-- see how well mv/count works
+explain
+select count(id) from sketch_input;
+
+explain
+select SketchToEstimate2(datatosketch2(id)) from sketch_input;
+select SketchToEstimate2(datatosketch2(id)) from sketch_input;
+
```

Comment:
I wonder about how we could do the rewite:
* earlier I wanted to rewrite `select count(distinct id)` to something like `select sketchtoestimate(datatosketch(id))` 
  but in case we don't have an (uptodate) mv to serve precomputed values for this it will go and work with sketches - do we still want it in that case as well? - I guess in some cases yes; calculating distinct in huge magnitudes could be highly resource intensive
* alternatively; we can add a rewrite for mv defitions with `select sketchtoestimate(datatosketch(id)) [...]` to `select count(distinct id)` - that way we only match those which do backed by mv-s